src/main/java/com/pagsestagio/movieapi/model/Filme.java

```diff
@@ -1,4 +1,86 @@
 package com.pagsestagio.movieapi.model;
 
-public record Filme(Integer identificador, String nomeFilme) {
-}
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+import java.util.Objects;
+import java.util.UUID;
+
+
+@Entity
+@Table(name = "MOVIE")
+public class Filme {
+    public Filme(Integer id, Integer idLegado, UUID idPublico, String nomeFilme) {
+        this.id = id;
+        this.idLegado = idLegado;
+        this.idPublico = UUID.randomUUID();
+        this.nomeFilme = nomeFilme;
+    }
+
+    public Filme() {
+        this.idPublico = UUID.randomUUID();
+    }
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "ID")
+    private Integer id;
+
+    @Column(name = "LEGACY_ID")
+    private Integer idLegado;
+
+    @Column(name = "PUBLIC_ID")
+    private UUID idPublico;
+
+    @Column(name = "MOVIE_NAME")
+    private String nomeFilme;
+
+    public Integer getId() {
+        return id;
+    }
+
+    public void setId(Integer id) {
+        this.id = id;
+    }
+
+    public Integer getIdLegado() {
+        return idLegado;
+    }
+
+    public void setIdLegado(Integer idLegado) {
+        this.idLegado = idLegado;
+    }
+
+    public UUID getIdPublico() {
+        return idPublico;
+    }
+
+    public void setIdPublico(UUID idPublico) {
+        this.idPublico = idPublico;
+    }
+
+    public String getNomeFilme() {
+        return nomeFilme;
+    }
+
+    public void setNomeFilme(String nomeFilme) {
+        this.nomeFilme = nomeFilme;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Filme filme = (Filme) o;
+        return Objects.equals(id, filme.id) && Objects.equals(idLegado, filme.idLegado) && Objects.equals(idPublico, filme.idPublico) && Objects.equals(nomeFilme, filme.nomeFilme);
```

Comment:
pensando bem seu equals tem um problema.
tipo, o que define num entidade se ela é a mesma que outra é o id.

porque imagina só. você tá lá o objeto que representa sualinha no banco.

{
 id=1
 idLegado=null, 
 idPublico=UUID
 nomeFilme="matrix"
}

se você for lá e alterar o matrix para Matrix
{
 id=1
 idLegado=null, 
 idPublico=UUID
 nomeFilme="Matrix"
}

você tá falando do mesmo registro. é só uma atualização.
mas se você comparar para ver se é o mesmo objeto isso vai dar como false.
e não deveria.

Ai a implementação deveria ser algo assim:

if(id != null && filme.id != null {
    Objects.equals(id, filme.id);
} else if( id == null && filme.id == null {
  return Objects.equals(idLegado, filme.idLegado) && Objects.equals(idPublico, filme.idPublico) && Objects.equals(nomeFilme, filme.nomeFilme);
} else {
 return false;
}

Reply:
mesma coisa no hashCode;

    @Override
    public int hashCode() {

        if(id != null {
           Objects.hash(id);
        }
        return Objects.hash(idLegado, idPublico, nomeFilme);
    }